### DISCUSSION TITLE
Use pyinfra instead of a Dockerfile to build docker images?

### DISCUSSION THREAD

**Discussion by dahjelle** [pyinfra-dev/pyinfra#1100] [Q&A]
Is it reasonable to use `pyinfra` to build docker images as a replacement for a `Dockerfile`?

(I'm a new to both, so I may be missing something obvious.)

I tried the [getting started example](https://docs.pyinfra.com/en/2.x/getting-started.html#create-a-deploy), removing `my-server.net` from the inventory. I also had to remove the `_sudo=True` line and add `apt.update()` into the `deploy.py` file. 

`pyinfra` finishes with the line:

```
    [@docker/ubuntu:18.04] docker build complete, image ID: d459fd5d33fa
```

But `docker image ls` doesn't show an image with that ID!

```bash
% docker image ls
REPOSITORY                TAG         IMAGE ID      CREATED         SIZE
<none>                    <none>      fc3d129d459f  27 seconds ago  58.9 MB
docker.io/library/ubuntu  18.04       d1a528908992  11 months ago   58.9 MB
```

(I was also expecting to be able to tag the image, as I would in the first line in a `Dockerfile`.)

It does appear that if I run the `fc3d129d459f` image that it is what I'm looking for.

I'm probably missing some basic concepts, but if someone has a moment to point me in the right direction, I'd appreciate it!

**Comment by Fizzadar** [comment]
I don't see any reason not to! I actually would like to go full circle and implement a connector that writes `Dockerfile`s based on operations, empty PR waiting for it: https://github.com/pyinfra-dev/pyinfra-dockerfile-connector
